src/GuessControl.jsx

```diff
@@ -1,47 +1,30 @@
-import React, { Component } from "react";
+import React, { Component, useState } from "react";
 import Button from "./Button";
 
-class GuessControl extends Component {
-  constructor(props) {
-    super(props);
+function GuessControl(props) {
+  const { onGuess } = props
 
-    this.state = {
-      currentGuess: "",
-    };
+  const [currentGuess, setCurrentGuess] = useState("")
 
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleInputChange = this.handleInputChange.bind(this);
-    this.onSubmitGuess = this.onSubmitGuess.bind(this);
+  const handleInputChange = (event) => {
+    setCurrentGuess(event.target.value)
   }
 
-  handleInputChange(event) {
-    this.setState({ currentGuess: event.target.value });
+  const onSubmitGuess = () => {
+    onGuess(parseInt(currentGuess))
```

Comment:
because it's a refactor and not a rewrite, you should maintain using `Number` instead of `parseInt` but for this use case it should work mostly the same.